File: crud/src/main/java/fr/cicco/crud/entity/Ingredient.java
package fr.cicco.crud.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Builder;
import lombok.NoArgsConstructor;    
    
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Builder
public class Ingredient   {
    @Id()
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private Double prix;
    @ManyToMany
    @JoinTable(name="ingredient_pizza",
            joinColumns = @JoinColumn(name="ingredient_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name="pizza_id", referencedColumnName = "id")
    )
    private List<Pizza> pizzaList;

}

File: crud/springCLI/python_entities_template/ingredient.py

from ..datas.autocompletion import *

name = 'Ingredient'

entity = {
    
    'Ingredient' : {
            
        'id':{
            Type: Long,
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'nom':{
            Type: String,
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'prix':{
            Type: Double,
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'pizzaList':{
            Type: "List<Pizza>",
            options:{
                'dto':[
                ]
        
            }
        },

        }

}


